drivers/ws281x/atmega.c

```diff
@@ -0,0 +1,206 @@
+                        : [port]    "I" (_SFR_IO_ADDR(PORTC)),
+                          [on]      "r" (mask_on),
+                          [off]     "r" (mask_off)
+                    );
+                    cnt--;
+                    data <<= 1;
+                }
+                pos++;
+            }
+            break;
+        case PORT_D:
+            while (pos < end) {
+                uint8_t cnt = 8;
+                uint8_t data = *pos;
+                while (cnt > 0) {
+                    __asm__ volatile(                       /* Cycles | 1 | 0 */
+                        "out    %[port], %[on]"     "\n\t"  /* 1      | x | x */
+                        "sbrs   %[data], 7"         "\n\t"  /* 1      | x | x */
+                        "out    %[port], %[off]"    "\n\t"  /* 1      | - | x */
+                        "rjmp   .+0"                "\n\t"  /* 2      | x | x */
+                        "out    %[port], %[off]"    "\n\t"  /* 1      | x | x */
+                        /* Total CPU cycles for high period:
+                         * 5 cycles for bit 1, 2 cycles for bit 0 */
+                        : [data]    "+r" (data)
+                        : [port]    "I" (_SFR_IO_ADDR(PORTD)),
+                          [on]      "r" (mask_on),
+                          [off]     "r" (mask_off)
+                    );
+                    cnt--;
+                    data <<= 1;
+                }
+                pos++;
+            }
+            break;
+        default:
+            assert(0);
+            break;
+    }
+#elif (CLOCK_CORECLOCK >= 15500000U) && (CLOCK_CORECLOCK <= 16500000U)
+    while (pos < end) {
+        uint8_t cnt = 8;
+        uint8_t data = *pos;
+        while (cnt > 0) {
+            __asm__ volatile(                       /* CPU Cycles | 1 | 0 */
+                "st     %a[port], %[on]"    "\n\t"  /* 2          | x | x */
+                "sbrc   %[data], 7"         "\n\t"  /* 1          | x | x */
+                "rjmp   .+0"                "\n\t"  /* 2          | x | - */
+                "sbrc   %[data], 7"         "\n\t"  /* 1          | x | x */
+                "rjmp   .+0"                "\n\t"  /* 2          | x | - */
+                "sbrc   %[data], 7"         "\n\t"  /* 1          | x | x */
+                "nop"                       "\n\t"  /* 1          | x | - */
+                "st     %a[port], %[off]"   "\n\t"  /* 2          | x | x */
+                /* Total CPU cycles for high period:
+                 * 10 cycles for bit 1, 5 cycles for bit 0 */
+                : [data]    "+r" (data),
+                  [port]    "=e" (port_addr)
+                : [on]      "r" (mask_on),
+                  [off]     "r" (mask_off)
+            );
+            cnt--;
+            data <<= 1;
+        }
+        pos++;
+    }
+#else
+#error "No low level WS281x implementation for ATmega CPUs for your CPU clock"
+#endif
+    xtimer_usleep(WS281X_T_END_US);
```

Comment:
This is already provided by the wrapper function now.

Reply:
Fixed